### PR TITLE
Use localstack to mock the SQS dependency

### PR DESCRIPTION
**Pull request by sunil-sadasivan**
One of the things that has been on the back of my mind when I switched from using sidekiq/redis to shoryuken/sqs was that we no longer have a smooth way to mock the queuing backend.  This makes for a suboptimal developer experience locally.  Prior to this change, devs needed to develop using a shared development SQS queue.  

This brings in the localstack container to the local setup and gives us a way to mock this dependency. 
https://github.com/localstack/localstack/ 

## To test

```
# bring up your stack including the localstack container
$ docker-compose up -d

# run the shoryuken worker
$ shoryuken start -q caseflow_development_high_priority caseflow_development_low_priority -R

# run the rails console in a separate term window
$ rails c
> HeartbeatTasksJob.perform_later
```
You should see the shoryuken job processing these messages. 

## Other Thoughts
The nice thing about localstack is it allows us to mock other AWS services.  Since we use lambda and s3, we may be able to use localstack to eventually replicate this locally as well (if we'd need to)

### CODE REVIEW

File: config/initializers/shoryuken.rb [hunk 1]
```diff
@@ -7,6 +7,16 @@
 ActiveJob::QueueAdapters::ShoryukenAdapter::JobWrapper
   .shoryuken_options(retry_intervals: [3.seconds, 30.seconds, 5.minutes, 30.minutes, 2.hours, 5.hours])
 
+if ENV['DEPLOY_ENV'] == 'local' or ENV['DEPLOY_ENV'] == 'development'
```
Comment (sunil-sadasivan) [inline config/initializers/shoryuken.rb]: @mdbenjam I'm not sure if this is the best way for me to run this only on local and development.  Let me know if you can think of a better approach...  For example, this won't work for the ssh forwarding environment

Comment (mdbenjam) [reply config/initializers/shoryuken.rb]: We could move this to the `environment.rb` files i.e. `development.rb`, `ssh_forwarding.rb`, and `local.rb`. Might be a little verbose but probably clearer.

Comment (sunil-sadasivan) [reply config/initializers/shoryuken.rb]: Ah, I like that better @mdbenjam, will do that instead.

**Comment by mdbenjam** [comment]
Wow this is really cool. I love it, and we should 💯 use this mock S3. We have a similar test bucket setup for our S3 tests, and then use our own fake'd S3 for all our tests that rely on S3. This would make things much cleaner!

**Comment by sunil-sadasivan** [comment]
@mdbenjam I added environment configs for specifying whether to override the sqs endpoint and create queues.  Would love a second pair of 👀

**Pull request merged**